libvips/iofuncs/source.c

```diff
@@ -153,7 +153,8 @@ vips_source_test_features( VipsSource *source )
 	 * vips_source_seek() etc. or we might trigger seek emulation.
 	 */
 	if( source->data ||
-		class->seek( source, 0, SEEK_CUR ) != -1 ) { 
+		(VIPS_CONNECTION( source )->descriptor != -1 &&
```

Comment:
I'm not sure here -- the subclass might not be using `descriptor`.

How about adding this test to `vips_source_seek_real()`, the default implementation? It could be eg.:

```C
    if( connection->descriptor != -1 )
        return( vips__seek_no_error( connection->descriptor, offset, whence ) );

    return( -1 );
```

Reply:
You're right, I forgot that for a moment. Resolved with https://github.com/kleisauke/libvips/commit/e437805fc19cdf9af89685576a08b3a502a2954d.

Do you think if `vips_source_read_real` needs something similar? I had a look, but I think there is always a file descriptor present when that function is called.

Reply:
Ooooof, I'm not sure. I *think* it's OK.